include/dfu/flash_img.h

```diff
@@ -20,6 +20,32 @@ struct flash_img_context {
 	struct stream_flash_ctx stream;
 };
 
+/**
+ * @brief Enum which defines all verification algorithms
+ *
+ * Algorithm list:
+ * - SHA-256
+ */
+enum flash_img_check_alg {
+	FLASH_IMG_CHECK_ALG_SHA256,
+};
```

Comment:
Continuing from https://github.com/zephyrproject-rtos/zephyr/pull/26789#discussion_r457407391, maybe instead of enum, define callback prototype that has to be provided by developer so that custom check procedures could be provided without modifications to base code.